src/SablierMerkleVCA.sol

```diff
@@ -82,61 +82,81 @@ contract SablierMerkleVCA is
-                endTime: params.timestamps.end,
+        // Check: campaign expiration is at least 1 week later than the vesting end time.
+        if (params.expiration < params.schedule.endTime + 1 weeks) {
+            revert Errors.SablierMerkleVCA_ExpirationTooEarly({
+                endTime: params.schedule.endTime,
                 expiration: params.expiration
             });
         }
 
-        _timestamp = params.timestamps;
+        _schedule = params.schedule;
     }
 
     /*//////////////////////////////////////////////////////////////////////////
                            USER-FACING CONSTANT FUNCTIONS
     //////////////////////////////////////////////////////////////////////////*/
 
     /// @inheritdoc ISablierMerkleVCA
-    function timestamps() external view override returns (MerkleVCA.Timestamps memory) {
-        return _timestamp;
+    function calculateClaimAmount(uint128 fullAmount) external pure returns (uint128) {
+        // TODO: implement
+        fullAmount;
+        return 0;
+    }
+
+    /// @inheritdoc ISablierMerkleVCA
+    function calculateForgoneAmount(uint128 fullAmount) external pure returns (uint128) {
+        // TODO: implement
+        fullAmount;
+        return 0;
+    }
+
+    /// @inheritdoc ISablierMerkleVCA
+    function getSchedule() external view override returns (MerkleVCA.Schedule memory) {
+        return _schedule;
     }
 
     /*//////////////////////////////////////////////////////////////////////////
                           INTERNAL NON-CONSTANT FUNCTIONS
     //////////////////////////////////////////////////////////////////////////*/
 
     /// @inheritdoc SablierMerkleBase
-    function _claim(uint256 index, address recipient, uint128 amount) internal override {
+    function _claim(uint256 index, address recipient, uint128 fullAmount) internal override {
         uint40 blockTimestamp = uint40(block.timestamp);
 
-        // Check: unlock start time is in the past.
-        if (_timestamp.start >= blockTimestamp) {
-            revert Errors.SablierMerkleVCA_ClaimNotStarted(_timestamp.start);
+        // Load the vesting schedule into memory to avoid multiple SLOADs.
+        MerkleVCA.Schedule memory vestingSchedule = _schedule;
+
+        // Check: schedule start time is in the past.
+        if (vestingSchedule.startTime >= blockTimestamp) {
+            revert Errors.SablierMerkleVCA_ClaimNotStarted(vestingSchedule.startTime);
         }
 
-        uint128 claimableAmount;
+        // Calculate the claim and foregone amount.
+        uint128 claimAmount;
+        uint128 forgoneAmount;
 
-        // Calculate the claimable amount.
-        if (_timestamp.end <= blockTimestamp) {
-            // If the unlock period has ended, the recipient can claim the full amount.
-            claimableAmount = amount;
+        if (vestingSchedule.endTime <= blockTimestamp) {
+            // If the vesting period has ended, the recipient can claim the maturity amount.
```

Comment:
The only place where `maturity` is used. tbh its not clear what it means.

Reply:
Will remove